### DISCUSSION TITLE
FR: Use folder name as unique identifier when importing assets

### DISCUSSION THREAD

**Discussion by rob-c-baker** [engram-design/FeedMe#905] [Ideas]
### Description

If there is a file with the same name in several different folders within an asset volume, it would make sense to be able to select the folder as a unique field to consider when deciding to update or create new.

### Additional info

- Plugin version: 4.1.2
- Craft version: 3.2.10

**Comment by rob-c-baker** [comment]
Further to this: Assets can import into specific sub-folders of a volume as specified an Asset feed.

However, when referencing the previously uploaded asset within an asset field for a feed importing into an Entry channel, Craft appears to find the previously uploaded asset in the sub-folder but it gets moved to the root / base dir of the volume, out of the sub-folder where it previously was.

Perhaps I am missing something, but in the case where an existing asset is referenced in an asset field on an Entry, it should be possible to define the sub-folder within the volume in the feed options (last screen: field mapping) in which to look for the pre-existing asset.

**Comment by rob-c-baker** [comment]
I have also been looking at the Craft CMS based event: EVENT_SET_FILENAME in relation to this.

https://docs.craftcms.com/v3/extend/updating-plugins.html#modifyassetfilename

It's looking like I cannot prefix a folder path to the filename at this stage, maybe another new event would be required. Something like EVENT_SET_FILE_PATH ?

**Comment by DavidKabelitz** [comment]
Is there a solution to this? I have the same issue or problem. Need to import tons of data where some files do have the same name but in different folders...how can i import and select the correct asset?

Asset Field of the entry needs to be one file not multiple.

How its at the moment:

**Setting "Keep both"**
- Import started, entry created and first time asset appears asset will be created and connected to the entry
- Second time at a new entry with an asset that has the same file name a new one will be created and renamed but the first one is selected at the entry

**Setting "Replace existing"**
- Just one asset will be created and overwritten at the second time of import

**Reply by rob-c-baker** [reply]
I had to work-around this - I was able to adjust filenames at source before they were imported as there were few enough duplicates to handle - about 100.

**Reply by DavidKabelitz** [reply]
ah ok, at the moment i also have a work-around. Two imports and a script. First import of the assets and rename it if its exists also i put the old id to the assets, second import of the entries without assets and also the old id, at least the script that goes to all entries, select the asset based on the old id and saves the entry ;)
